pkgs/applications/networking/instant-messengers/feishu/default.nix

```diff
@@ -146,21 +150,29 @@ stdenv.mkDerivation rec {
     mkdir -p $out
     mv usr/share $out/
     mv opt/ $out/
-    chmod -R g-w $out
 
     substituteInPlace $out/share/applications/bytedance-feishu.desktop \
       --replace /usr/bin/bytedance-feishu-stable $out/opt/bytedance/feishu/bytedance-feishu
 
-    wrapProgram $out/opt/bytedance/feishu/feishu \
-      --prefix XDG_DATA_DIRS : "$XDG_ICON_DIRS:$GSETTINGS_SCHEMAS_PATH" \
-      --prefix LD_LIBRARY_PATH : ${rpath}:$out/opt/bytedance/feishu
+    # Wrap feishu and vulcan
+    # Feishu is the main executable, vulcan is the builtin browser
+    for executable in $out/opt/bytedance/feishu/{feishu,vulcan/vulcan}; do
+      wrapProgram $executable \
+        --prefix XDG_DATA_DIRS    :  "$XDG_ICON_DIRS:$GSETTINGS_SCHEMAS_PATH" \
+        --prefix LD_LIBRARY_PATH  :  ${rpath}:$out/opt/bytedance/feishu:${addOpenGLRunpath.driverLink}/share \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--enable-features=UseOzonePlatform --ozone-platform=wayland}}" \
+        --add-flags ${lib.escapeShellArg commandLineArgs}
```

Comment:
Maybe you should add something preventing the "empty string" corner case.

Something like `lib.optionalString (commandlineArgs != "") "add-flags ..."`

Reply:
Changed to `${lib.optionalString (commandLineArgs!="") "--add-flags ${lib.escapeShellArg commandLineArgs}"}`, now the "empty string" case is handled.